packages/dev/core/src/Misc/deepCopier.ts

```diff
@@ -13,6 +13,8 @@ const CloneValue = (source: any, destinationObject: any) => {
         return source.clone(destinationObject);
     } else if (source.clone) {
         return source.clone();
+    } else if (Array.isArray(source)) {
+        return source.slice();

```

Comment:
i assume a deep copy should also copy the content of the array (if they are not primitives). In what case does this happen?

Reply:
DeepCopier didn't support deep copying arrays at all. I use DeepCopier only to copy primitive arrays so I implemented it for it's needs as simple and fast as possible. I propose to leave it this way until someone needs to use DeepCopier to copy object arrays and leave to him/her to implement the functionality.